iocuddle-sgx/src/lib.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright 2019 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![deny(clippy::all)]
+#![allow(clippy::identity_op)]
+
+use std::marker::PhantomData;
+
+use bitflags::bitflags;
+use iocuddle::*;
+use sgx_types::{page, secs, sig};
+
+const SGX: Group = Group::new(0xA4);
+pub const ENCLAVE_CREATE: Ioctl<Write, &Create> = unsafe { SGX.write(0x00) };
+pub const ENCLAVE_ADD_PAGES: Ioctl<WriteRead, &AddPages> = unsafe { SGX.write_read(0x01) };
+pub const ENCLAVE_INIT: Ioctl<Write, &Init> = unsafe { SGX.write(0x02) };
+pub const ENCLAVE_SET_ATTRIBUTE: Ioctl<Write, &SetAttribute> = unsafe { SGX.write(0x03) };
```

Comment:
this seems unused

Reply:
Correct. It is defined in the kernel but we don't currently have any use for it and haven't investigated what it does yet.

Reply:
I see. I don't fully understand the mechanism behind these constants yet, but somehow I'm looking for a way to define in the public API fo this crate that this is not yet handled here.

Reply:
I have commented this out in the latest patches.